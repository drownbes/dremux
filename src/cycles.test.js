const { Graph } = require('./graph');
const { createSubGraph, leastIndexScc, simpleCycles } = require('./cycles');
const { tarjanScc } = require('./scc');

describe('cycles', () => {


  test('createSubGraph', () => {
    const g = new Graph();

    g.addEdge(0, 1);
    g.addEdge(1, 2);
    g.addEdge(2, 3);
    g.addEdge(3, 1);

    expect(createSubGraph(1, g)).toMatchSnapshot();
    expect(createSubGraph(2, g)).toMatchSnapshot();

  });

  test('leastIndexScc', () => {
    const g = new Graph();

    g.addEdge(0, 1);
    g.addEdge(1, 2);
    g.addEdge(2, 3);
    g.addEdge(3, 1);

    let sccs = tarjanScc(g);

    let ls = leastIndexScc(sccs, g);
    expect(ls).toMatchSnapshot();

  });

  test('simpleCycles', ()=> {
    const graph = new Graph();

    graph.addEdge(1, 2);
    graph.addEdge(1, 8);
    graph.addEdge(1, 5);
    graph.addEdge(2, 9);
    graph.addEdge(2, 7);
    graph.addEdge(2, 3);
    graph.addEdge(3, 1);
    graph.addEdge(3, 2);
    graph.addEdge(3, 6);
    graph.addEdge(3, 4);
    graph.addEdge(6, 4);
    graph.addEdge(4, 5);
    graph.addEdge(5, 2);
    graph.addEdge(8, 9);
    graph.addEdge(9, 8);

    expect(
      simpleCycles(graph)
    ).toMatchSnapshot();

  });

  test('simpleCycles', ()=> {
    const g = new Graph();

    g.addEdge(0, 1);
    g.addEdge(1, 2);
    g.addEdge(2, 3);
    g.addEdge(3, 1);
    g.addEdge(3, 4);
    g.addEdge(4, 5);
    g.addEdge(5, 3);

    expect(
      simpleCycles(g)
    ).toMatchSnapshot();

  });



});