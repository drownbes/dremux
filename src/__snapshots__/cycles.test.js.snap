// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cycles createSubGraph 1`] = `
Graph {
  "_data": Object {
    "1": Array [
      "2",
    ],
    "2": Array [
      "3",
    ],
    "3": Array [
      "1",
    ],
  },
  "vertexParams": Object {},
}
`;

exports[`cycles createSubGraph 2`] = `
Graph {
  "_data": Object {
    "2": Array [
      "3",
    ],
    "3": Array [],
  },
  "vertexParams": Object {},
}
`;

exports[`cycles leastIndexScc 1`] = `
Graph {
  "_data": Object {
    "1": Array [
      "2",
    ],
    "2": Array [
      "3",
    ],
    "3": Array [
      "1",
    ],
  },
  "minVertex": "1",
  "vertexParams": Object {},
}
`;

exports[`cycles simpleCycles 1`] = `
Array [
  Array [
    "1",
    "2",
    "3",
    "1",
  ],
  Array [
    "1",
    "5",
    "2",
    "3",
    "1",
  ],
  Array [
    "2",
    "3",
    "2",
  ],
  Array [
    "2",
    "3",
    "6",
    "4",
    "5",
    "2",
  ],
  Array [
    "2",
    "3",
    "4",
    "5",
    "2",
  ],
  Array [
    "8",
    "9",
    "8",
  ],
]
`;

exports[`cycles simpleCycles 2`] = `
Array [
  Array [
    "1",
    "2",
    "3",
    "1",
  ],
  Array [
    "3",
    "4",
    "5",
    "3",
  ],
]
`;
